.github/workflows/black.yml

```diff
@@ -0,0 +1,32 @@
+# GitHub Action that uses Black to reformat the Python code in an incoming pull request.
+# If all Python code in the pull request is compliant with Black then this Action does nothing.
+# Othewrwise, Black is run and its changes are committed back to the incoming pull request.
+# https://github.com/cclauss/autoblack
+# Thanks to the PSF!
+# https://github.com/psf/black/actions/runs/17913292/workflow
+
+name: autoblack
+on: [pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install Black
+        run: pip install black
+      - name: Run black --check .
+        run: black --check .
+      - name: If needed, commit black changes to the pull request
+        if: failure()
+        run: |
+          black .
+          git config --global user.name 'autoblack'
+          git config --global user.email 'cclauss@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+          git checkout $GITHUB_HEAD_REF
+          git commit -am "fixup: Format Python code with Black"
```

Comment:
fixup has a specific meaning in git commits so it might just be worth removing that part https://www.mikulskibartosz.name/git-fixup-explained/

Reply:
It was intentional to use that specific meaning.

Reply:
Ah, so it is intended that people will rebase the fixup?

Reply:
It was intended that if contributors rebase or maintainers squash and merge that the formatting commits disappear.

Reply:
That is rather handy actually. Thanks for thinking of that :)